src/components/Category.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+const Category = ({ registers, name }) => {
+  const [showButton, setShowButton] = useState(false);
+
+  const RegisterCategory = ({ regName }) => (
+    <button type="submit" onClick={() => setShowButton(!showButton)}>
+      {regName}
+    </button>
+  );
+
+  RegisterCategory.propTypes = {
+    regName: PropTypes.string.isRequired,
+  };
+
+  if (!showButton) {
+    return (
+      <div>
```

Comment:
is this div necessary? could we just return the RegisterCategory directly?

Reply:
I agree, Ill change it so it returns just a RegisterCategory